src/main/java/kayu/service/TaskList.java

```diff
@@ -0,0 +1,145 @@
+package kayu.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import kayu.exception.DukeException;
+import kayu.exception.StorageException;
+import kayu.task.Task;
+
+/**
+ * Manages tasks held by the {@link kayu.Kayu}.
+ */
+public class TaskList {
+
+    // Error message formats.
+    protected static final String INVALID_TASK_ERROR_MESSAGE = "Task number '%d' is invalid.";
+    protected static final String FULL_CAPACITY_ERROR_MESSAGE = "Unable to execute as list is full.";
+    protected static final String EMPTY_LIST_ERROR_MESSAGE = "Unable to execute as list is empty.";
+
+    protected static final int MAX_STORAGE = 100;
```

Comment:
Not sure if the max size is necessary as the point of the arrayList is so that it can grow